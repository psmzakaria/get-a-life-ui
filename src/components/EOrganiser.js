import React from "react";
import { Card, Divider, Header } from "semantic-ui-react";
import GuestModal from "./GuestModal";
import HostModal from "./HostModal";
import Logo from "./Logo";

const getGuestModal = (event, status) => {
  return <GuestModal key={event._id} event={event} status={status} />;
};

const getHostModal = (event, status) => {
  return <HostModal key={event._id} event={event} status={status} />;
};

const EOrganiser = props => {
  return (
    <div className="eorganiser-container">
      <Logo />
      <h2> Your Hosted Events</h2>
      {props.hostedEvents.length > 0 && (
        <Card.Group>
          {props.hostedEvents.map((event, index) => {
            return getHostModal(event, props.statuses[index]);
          })}
        </Card.Group>
      )}
      {props.hostedEvents.length === 0 && (
        <Header as="h2" textAlign="center" disabled>
          You have no hosted events
        </Header>
      )}
      <Divider />
      <h2> Your Invited Events</h2>
      {props.invitedEvents.length > 0 && (
        <Card.Group>
          {props.invitedEvents.map((event, index) => {
            return getGuestModal(
              event,
              props.statuses[index],
              props.getUserData,
              props.username
            );
          })}
        </Card.Group>
      )}
      {props.invitedEvents.length === 0 && (
        <Header as="h2" textAlign="center" disabled>
          You have been invited
        </Header>
      )}
      <Divider />
      <h2> You have a life</h2>
      {props.acceptedEvents.length > 0 && (
        <Card.Group>
          {props.acceptedEvents.map((event, index) => {
            return getGuestModal(event, props.statuses[index]);
          })}
        </Card.Group>
      )}
      {props.acceptedEvents.length === 0 && (
        <Header as="h2" textAlign="center" disabled>
          Soon ...
        </Header>
      )}
    </div>
  );
};

export default EOrganiser;
