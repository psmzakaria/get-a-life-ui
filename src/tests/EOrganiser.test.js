import React from "react";
import EOrganiser from "./../components/EOrganiser";

import ShallowRenderer from "react-test-renderer/shallow";

it("renders the create component of the app ", () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <EOrganiser hostedEvents={[]} statuses={[]} invitedEvents={[]} />
  );
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
