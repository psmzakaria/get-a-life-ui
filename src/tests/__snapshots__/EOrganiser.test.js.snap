// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the create component of the app  1`] = `
<div
  className="eorganiser-container"
>
  <h2>
     Your Hosted Events
  </h2>
  <Header
    as="h2"
    disabled={true}
    textAlign="center"
  >
    You have no hosted events
  </Header>
  <Divider />
  <h2>
     Your Invited Events
  </h2>
  <Header
    as="h2"
    disabled={true}
    textAlign="center"
  >
    You have been invited
  </Header>
  <Divider />
  <h2>
     You have a life
  </h2>
  <Header
    as="h2"
    disabled={true}
    textAlign="center"
  >
    Soon ...
  </Header>
</div>
`;
